pkg/cvo/availableupdates.go

```diff
@@ -126,11 +132,19 @@ func (optr *Operator) syncAvailableUpdates(ctx context.Context, config *configv1
 		optrAvailableUpdates.UpdateService = updateService
 		optrAvailableUpdates.Channel = channel
 		optrAvailableUpdates.Architecture = desiredArch
-		optrAvailableUpdates.Current = current
-		optrAvailableUpdates.Updates = updates
-		optrAvailableUpdates.ConditionalUpdates = conditionalUpdates
 		optrAvailableUpdates.ConditionRegistry = optr.conditionRegistry
 		optrAvailableUpdates.Condition = condition
+
+		responseFailed := (condition.Type == configv1.RetrievedUpdates &&
+			condition.Status == configv1.ConditionFalse &&
+			(condition.Reason == "RemoteFailed" ||
+				condition.Reason == "ResponseFailed" ||
+				condition.Reason == "ResponseInvalid"))
+		if !responseFailed || (responseFailed && !preserveCacheOnFailure) {
+			optrAvailableUpdates.Current = current
+			optrAvailableUpdates.Updates = updates
+			optrAvailableUpdates.ConditionalUpdates = conditionalUpdates
+		}
```

Comment:
Several thoughts about this code:

1. the `condition.Type == configv1.RetrievedUpdates` check seems misleading here. `condition` comes from `calculateAvailableUpdatesStatus` which I'd expect to always return that condition type. If we receive a different type here I'd expect that to mean "this looks like a bug, likely someone messed with the code?", not "failed to get recommendations".

2. Shall we interpret `Status=Unknown` as `!responseFailed` (== success)? Shouldn't we look for explicit success rather than for explicit failure?

3. We introduce two booleans with that we end up using negated. How about inverting `preserveCacheOnFailure` into `clearCacheOnFailure`. We'd need to initialize it to true and make it `false` only in last two condition branches. We can then invert `responseFailed` into `responseSucceeded` by negating the full expression in its assignment. This way, the condition on L143 would become `if responseSucceeded || (!responseSucceeded && clearCacheOnFailure)` which alone is IMO better readable, but the negated responseSucceed is actually redundant (we only enter the second `||` clause on `!responseSucceeded`)  and we could simplify to `if responseSucceeded || clearCacheOnFailure` which is IMO very straightforward and understandable.

Not blocking on these, the existing code is good but leaving this here for consideration.

Reply:
Yeah, makes sense. I was aware of the nuances you mention. No problem in merging this as is.

Reply:
/lgtm

;)